apis/cluster/v1alpha1/types.go

```diff
@@ -24,9 +24,11 @@ import (
 
 // ClusterParameters define the desired state of an ArgoCD Cluster
 type ClusterParameters struct {
-	// Server is the API server URL of the Kubernetes cluster
+	// Server is the API server URL of the Kubernetes cluster. Optional if using a kubeconfig
+	// +optional
 	Server string `json:"server"`
-	// Name of the cluster. If omitted, will use the server address
+	// Name of the cluster. If omitted, will use the server address. Optional if using a kubeconfig
+	// +optional
 	Name string `json:"name"`
```

Comment:
Optional fields should be declared as pointers.

Reply:
Converted to a pointer.